code/game/machinery/teleporter.dm

```diff
@@ -259,23 +248,25 @@
 		src.to_bump(AM)
 		return
 	spawn()
-		if (src.engaged)
-			teleport(AM)
+		if (src.engaged && teleport(AM))
 			use_power(5000)
 
 /obj/machinery/teleport/hub/proc/teleport(atom/movable/M as mob|obj)
-	var/atom/l = src.loc
-	var/obj/machinery/computer/teleporter/com = locate(/obj/machinery/computer/teleporter, locate(l.x - 2, l.y, l.z))
+	var/obj/machinery/teleport/station/st = locate(/obj/machinery/teleport/station, orange(1))
```

Comment:
I'm not sure if it matters that much, but I belive `locate(x) in orange(1)` would be clearer and achieve the same effect.
Same for the line below.